storage/postgres/tree_storage.go

```diff
@@ -141,6 +144,13 @@ func (p *pgTreeStorage) getStmt(ctx context.Context, skeleton *statementSkeleton
 	}
 
 	statement, err := expandPlaceholderSQL(skeleton)
+
+	counter := skeleton.restPlaceholders*skeleton.num + 1
```

Comment:
This is a different calculation than `expandPlaceholderSQL()` uses to determine how high its counter will get, which could cause a bug if the skeleton SQL is changed in the future. Perhaps move this code into `expandPlaceholderSQL()` and handle "<param>" there?

Reply:
I plan on revamping the expandPlaceholderSQL to bring it and the subsequent code into the same place.  The problem I ran into was this:  Expand placeholder only gives the user two choices for dynamic sql creation, one method for the first replace and another standard for all subsequent replacement.  I found that, if the query had any single parameters in need of replacing after the expandPlaceholderSQL was run, this couldn't be used. Since the Postgres statements need variable replacement with $<num> rather than ?, I needed to know the last value from this call to correctly build the sql.  It is unwieldy and I will streamline it, but I couldn't get these queries to work without these updates.